docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml

```diff
@@ -225,110 +183,53 @@
-    - block:
-      - name: Get armada pod name
-        command: >-
-          kubectl get pods -n armada -o custom-columns=NAME:metadata.name --no-headers
-        environment:
-          KUBECONFIG: "/etc/kubernetes/admin.conf"
-        register: armada_pod
-
-      - name: Show armada pod
-        debug:
-          msg:
-          - "armada pod ID: {{ armada_pod.stdout }}"
-
-      - name: Copy files into tiller container if using helmv2
-        command: "{{ item }}"
-        with_items:
-          - kubectl -n armada cp
-            {{ manager_chart }}
-            {{ armada_pod.stdout }}:/tmp/.
-            -c tiller
-          - kubectl -n armada cp
-            {% if helm_chart_overrides is defined %}{{ helm_chart_overrides }}{% endif %}
-            {{ armada_pod.stdout }}:/tmp/.
-            -c tiller
-        environment:
-          KUBECONFIG: "/etc/kubernetes/admin.conf"
-
-      - name: Reinstall Deployment Manager (helmv2)
-        command: >-
-          /usr/local/sbin/helmv2-cli -- helm upgrade
-          --install deployment-manager
-          --values /tmp/{{ helm_chart_overrides | basename }}
-          /tmp/{{ manager_chart | basename }}
-        environment:
-          KUBECONFIG: "/etc/kubernetes/admin.conf"
-
-      when: helmv2_installed and helmv2_dm_exists.rc == 0
+    # DM system app helm check, upload/apply helm-chart
+    - block:  
+        # Find dm-system app helm-chart
+       - name: Find deployment-manager app helm-chart in application directory
+         find:
+          paths: /usr/local/share/applications/helm/
+          patterns: 'deployment-manager*.tgz'
+         register: app_helm_chart
+
+       - name: Fail deployment-manager app helm-chart does not exists
+         fail:
+           msg: "DM system app helm-chart does not exists in application directory"
+         when: (app_helm_chart.files[0].path | length == 0)
+   
+        # dm-system app upload/apply
+       - name: Upload deployment-manager app Helm Chart
+         shell: source /etc/platform/openrc; system application-upload {{ manager_app_chart }}
+
+       - name: Wait deployment-manager app to become uploaded
+         shell: source /etc/platform/openrc; system application-list | grep deployment-manager | awk '{ print $10 }'
+         register: app_upload_status
+         retries: 3
+         delay: 6
+         until: ("uploaded" in app_upload_status.stdout)
+
+       - name: Fail deployment-manager app upload fails
+         fail:
+           msg: "The upload of deployment-manager system app has failed."
+         when: ("upload-failed" in app_upload_status.stdout)
+
+       - name: Apply deployment-manager app
+         shell: source /etc/platform/openrc; system application-apply deployment-manager   
+         when: ("uploaded" in app_upload_status.stdout)
+
+       - name: Wait deployment-manager app to become applied
+         shell: source /etc/platform/openrc; system application-list | grep deployment-manager | awk '{ print $10 }'
```

Comment:
Please use `--nowrap` with `system` commands otherwise it may be prone to unexpected failures.
`source /etc/platform/openrc; system application-list --nowrap | grep deployment-manager | awk '{ print $10 }'`